### PR TITLE
xds server: Update reference/resource checks for DefaultFilterChain

### PR DESCRIPTION
**Pull request by sfc-gh-randerson**
This updates the consistency check to also consider the "DefaultFilterChain" in addition to the list of filter chains.

### CODE REVIEW

File: pkg/cache/v3/resource.go [hunk 1]
```diff
@@ -202,22 +202,11 @@ func getListenerReferences(src *listener.Listener, out map[resource.Type]map[str
...
-			}
+		getListenerReferencesFromChain(chain, routes)
+	}
 
-			// If the scoped route mapping is embedded, add the referenced route resource names.
-			for _, s := range config.GetScopedRoutes().GetScopedRouteConfigurationsList().GetScopedRouteConfigurations() {
-				routes[s.GetRouteConfigurationName()] = true
-			}
-		}
+	if src.DefaultFilterChain != nil {
```
Comment (valerian-roche) [inline pkg/cache/v3/resource.go:208]: Minor: the current code does not make an assumption on src not being `nil`, whereas this change would panic. Can you use the protobuf safe getter for consistency?

**Review by valerian-roche — COMMENTED** [review]
Minor comment only
Tbh this code is not really maintained and very likely out of sync with envoy which has evolved on supporting partial snapshots. 
Can you confirm if you rely on this functionality? My long term goal is to fully remove this logic as it is imo out of place and is also creating issues to support other data-planes (for instance grpc libraries)

File: pkg/cache/v3/resource.go (hunk 1 above)
Comment (sfc-gh-randerson) [reply pkg/cache/v3/resource.go]: Ah, good point. Fixed.

**Comment by sfc-gh-randerson** [comment]
> Minor comment only Tbh this code is not really maintained and very likely out of sync with envoy which has evolved on supporting partial snapshots. Can you confirm if you rely on this functionality? My long term goal is to fully remove this logic as it is imo out of place and is also creating issues to support other data-planes (for instance grpc libraries)

By "this code", do you mean the resource/reference check part of cache.Snapshot? Or something more broad?

We are setting ads=true when we call NewSnapshotCache() which seems to imply that our snapshots have to be consistent, which is what prompted this change. We can probably change away from that, but there's not really an indication right now that it's discouraged.

**Review by valerian-roche — APPROVED** [review]
lgtm

**Pull request merged**